pilot/pkg/config/kube/ingress/controller_test.go

```diff
@@ -26,15 +26,15 @@ import (
 	"istio.io/istio/pilot/pkg/model"
 	kubecontroller "istio.io/istio/pilot/pkg/serviceregistry/kube/controller"
 	"istio.io/istio/pkg/config"
-	"istio.io/istio/pkg/config/mesh"
+	"istio.io/istio/pkg/config/mesh/meshwatcher"
 	"istio.io/istio/pkg/config/schema/gvk"
 	"istio.io/istio/pkg/kube"
 	"istio.io/istio/pkg/kube/kclient/clienttest"
 	"istio.io/istio/pkg/util/sets"
 )
 
 func newFakeController() (model.ConfigStoreController, kube.Client) {
-	meshHolder := mesh.NewTestWatcher(&meshconfig.MeshConfig{
+	meshHolder := meshwatcher.NewFixedWatcher(&meshconfig.MeshConfig{
```

Comment:
Maybe I'll see it as I get further down, but I'm curious if there's a diff between TestWatcher and FixedWatcher

Reply:
TestWatcher waited until the mesh config was sent through the handlers. It did this by utilizin the fact handlers were called in order and blocked each other before -- krt doesn't treat them that way and each handler is independant, which makes the pattern not work.

... but none of the tests actually depended on the waiting anyways ,so I just killed it

Reply:
No issues from me; makes sense